tests/contracts/main/staking_escrow/test_staking_escrow.py

```diff
@@ -170,3 +174,371 @@ def test_slashing(testerchain, token, worklock, escrow):
+    merged = escrow.functions.getFlags(staker2).call()
+    assert not merged
+
+    # Request can be done only with the same operator
+    with pytest.raises((TransactionFailed, ValueError)):
+        tx = threshold_staking.functions.requestMerge(staker2, operator1).transact()
+        testerchain.wait_for_receipt(tx)
+
+
+def test_confirm_merge(testerchain, threshold_staking, escrow):
+    creator, staker, operator = testerchain.client.accounts[0:3]
+    merge_confirmations_log = escrow.events.MergeConfirmed.createFilter(fromBlock='latest')
+    staking_error = NU(10**15, "NuNit").to_nunits()
+
+    # Can't confirm merge without request
+    with pytest.raises((TransactionFailed, ValueError)):
+        tx = escrow.functions.confirmMerge(staker).transact()
+        testerchain.wait_for_receipt(tx)
+
+    # Prepare staker and make a request
+    value = NU(1000, "NU").to_nunits()
+    tx = escrow.functions.setStaker(staker, value, 0).transact()
+    testerchain.wait_for_receipt(tx)
+    tx = threshold_staking.functions.requestMerge(staker, operator).transact()
+    testerchain.wait_for_receipt(tx)
+
+    # Emulate unstake
+    tx = threshold_staking.functions.setStakedNu(operator, value - staking_error - 1).transact()
+    testerchain.wait_for_receipt(tx)
+
+    # Must be staked everything to allow confirm
+    with pytest.raises((TransactionFailed, ValueError)):
+        tx = escrow.functions.confirmMerge(staker).transact()
+        testerchain.wait_for_receipt(tx)
+
+    # And everything must be authorized
+    tx = threshold_staking.functions.requestMerge(staker, operator).transact()
+    testerchain.wait_for_receipt(tx)
+    tx = threshold_staking.functions.setStakedNu(operator, value - staking_error).transact()
+    testerchain.wait_for_receipt(tx)
+    with pytest.raises((TransactionFailed, ValueError)):
+        tx = escrow.functions.confirmMerge(staker).transact()
+        testerchain.wait_for_receipt(tx)
+
+    tx = threshold_staking.functions.setMinStaked(operator, value - staking_error).transact()
+    testerchain.wait_for_receipt(tx)
+    tx = escrow.functions.confirmMerge(staker).transact()
+    testerchain.wait_for_receipt(tx)
+    merged = escrow.functions.getFlags(staker).call()
+    assert merged
+
+    events = merge_confirmations_log.get_all_entries()
+    assert len(events) == 1
+    event_args = events[-1]['args']
+    assert event_args['staker'] == staker
+
+    # Confirm can be made only once
+    with pytest.raises((TransactionFailed, ValueError)):
+        tx = escrow.functions.confirmMerge(staker).transact()
+        testerchain.wait_for_receipt(tx)
+
+
+def test_withdraw(testerchain, token, worklock, threshold_staking, escrow):
+    creator, staker, operator = testerchain.client.accounts[0:3]
+    withdrawal_log = escrow.events.Withdrawn.createFilter(fromBlock='latest')
+
+    # Deposit some tokens
+    value = NU(ONE_HOUR, 'NU').to_nunits()  # Exclude rounding error
```

Comment:
I just need some number that will be divided by ONE_HOUR without remainder. Should I add one more variable to make it more clear?

Reply:
👍 hehe yeah I figured. Instead of a variable a comment could suffice - but up to you, I'm fine either way. it just jumped out at me, but not a problem.